core/web/keys_controller.go

```diff
@@ -67,6 +73,21 @@ func (kc *keysController[K, R]) Create(c *gin.Context) {
 		jsonAPIError(c, http.StatusInternalServerError, err)
 		return
 	}
+
+	// Emit audit log, determine if Terra or Solana key
+	switch unwrappedKey := any(key).(type) {
+	case terrakey.Key:
```

Comment:
Ah, we still need to find a different way to handle these different cases, since enumerating them internally defeats the purpose of the generic code. It looks like interfaces should be able to do most of the work for the data, if we pass in just the set of event ids instead. I can help if the generics are confusing.

Reply:
I think we can still handle this in a follow up. I believe @CL-Andrew has a shortcut task for it.

Reply:
I disagree. This is a blocker. It is already broken, and doesn't show until runtime.